src/main/controller/WorkoutController.ts

```diff
@@ -0,0 +1,82 @@
+import { Request, Response } from 'express';
+import db from '../models';
+import { Route } from '../constant/Route';
+
+export class WorkoutController {
+  getAllWorkouts = async (_req: Request, res: Response): Promise<any> => {
+    try {
+      const workouts = await db.Workout.findAll();
+      return res.status(200).send(workouts);
+    } catch (e) {
+      return res.status(400).send({ e });
+    }
+  };
+
+  getAllWorkoutsMinimal = async (
+    _req: Request,
+    res: Response
+  ): Promise<any> => {
+    try {
+      const workouts = await db.Workout.findAll({
+        attributes: [
+          'name',
+          'totalWorkoutTime',
+          'creationDate',
+          'lastModificationDate',
+        ],
+      });
+      return res.status(200).send(workouts);
+    } catch (e) {
+      return res.status(400).send({ e });
+    }
+  };
+
+  getWorkoutMinimal = async (req: Request, res: Response) => {
+    try {
+      if (!req.params.workoutId) {
+        throw `Missing workoutId param. Endpoint usage: /${Route.WORKOUTS}/minimal/:workoutId`;
+      }
+
+      const workout = await db.Workout.findOne({
+        attributes: [
+          'name',
+          'totalWorkoutTime',
+          'creationDate',
+          'lastModificationDate',
+        ],
+        where: { id: req.params.workoutId },
+      });
+
+      return res.status(200).send(workout);
+    } catch (e) {
+      return res.status(400).send({ e });
+    }
+  };
+
+  addWorkout = async (req: Request, res: Response) => {
+    try {
+      const newWorkout = await db.Workout.create(req.body);
```

Comment:
Could we do some validation on the req.body, probably easiest through creating a 'Workout' model and verifying that everything is in the expected format?